File: java/src/main/java/pizza/controller/TimeController.java
package pizza.controller;

import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;
import pizza.Time;
import pizza.repository.TimeRepository;

@RestController
public class TimeController {
    private TimeRepository timeRepository;

    public TimeController(TimeRepository timeRepository) {
        this.timeRepository = timeRepository;
    }

    @RequestMapping("/time")
    @ResponseBody
    public Time getTime() {
        String currentTime = timeRepository.getTime();

        return new Time(currentTime);
    }

}

File: kotlin-mvc/src/main/kotlin/com/hyvee/application/controllers/TimeController.kt
package com.hyvee.application.controllers

import com.hyvee.application.domain.Time
import com.hyvee.application.repositories.TimeRepository
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class TimeController(
    private val timeRepository: TimeRepository
) {
    @GetMapping("/time")
    fun getTime() = timeRepository.getTime()?.let {
        Time(it)
    }
}
